protocols/delegate-manager/contracts/DelegateManager.sol

```diff
@@ -0,0 +1,121 @@
+/*
+  Copyright 2019 Swap Holdings Ltd.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity 0.5.10;
+pragma experimental ABIEncoderV2;
+
+import "@airswap/indexer/contracts/interfaces/IIndexer.sol";
+import "@airswap/swap/contracts/interfaces/ISwap.sol";
+import "@airswap/delegate-factory/contracts/interfaces/IDelegateFactory.sol";
+import "@airswap/delegate/contracts/interfaces/IDelegate.sol";
+import "@airswap/types/contracts/Types.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+contract DelegateManager is Ownable {
+
+    IDelegateFactory public factory;
+    IDelegate public delegate;
+
+    /**
+      * @notice Constructor to create the Delegate Manager
+      * @dev factory is unable to be changed after the DelegateManager is constructed
+      * @param _factory address of the factory contract that will create the delegates
+      * @param _tradeWallet the wallet that the delegate will be trading on behalf of
+      */
+    constructor(IDelegateFactory _factory, address _tradeWallet) public {
+        factory = _factory;
+        delegate = IDelegate(factory.createDelegate(msg.sender, _tradeWallet));
```

Comment:
why not just leave the DelegateManager as the owner of the Delegate so you dont need admin roles?

Reply:
that'd doable. Since the original design has deviated quite a bit with regard to having a manager that manages many things, do you think it makes more sense to move the `setRuleAndIntent()` and `unsetRuleAndIntent()` methods into the Delegate?